src/OpenTelemetry.AutoInstrumentation/Configurations/ResourceDetector.cs

```diff
@@ -0,0 +1,38 @@
+// <copyright file="ResourceDetector.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.AutoInstrumentation.Configurations;
+
+/// <summary>
+/// Enum representing supported resource detectors.
+/// </summary>
+internal enum ResourceDetector
+{
+    /// <summary>
+    /// Environmental variables resource detector.
+    /// </summary>
+    EnvironmentalVariables = 0,
+
+    /// <summary>
+    /// TelemetrySdk resource detector.
+    /// </summary>
+    TelemetrySdk = 1,
```

Comment:
I think that these two should be always enabled.

Reply:
> That would require custom plugin then to override this behavior.

Exactly. Is there any reason why these should not be always enabled? Let me share my reasoning.

I think the env var detector is critical for auto-instrumentation as this is the recommended way to add additional resources. Otherwise, we would need to document that [this](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/docs/config.md#resources) section relies on this detector and has to be enabled. I feel that these env vars should always be supported.

Regarding TelemetrySDK  [the specification](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/resource/semantic_conventions#telemetry-sdk) it is stable and I am not sure if there is a good use case to disable it.

I believe a common setup would be to set `OTEL_DOTNET_AUTO_RESOURCE_DETECTOR_ENABLED=false` to rely on [Collector's Resource Detection Processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/resourcedetectionprocessor)

Reply:
> Regarding TelemetrySDK [the specification](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/resource/semantic_conventions#telemetry-sdk) it is stable and I am not sure if there is a good use case to disable it.

TelemetrySDK attributes seem to be required.

Reply:
Related to Env.Vars. and TelemetrySdk. Both of them are enabled by default.

Now I am introducing new layer of configuration. It is not different than call SetReourceBuilde(ResourceBuilder.CreateEmpty()). As I remember there is also possibility to manage resource on container side. So the users can drop them there.

For sure, I can remove this two options from the configuration but I do not see any good reason to do it.

Reply:
> I believe a common setup would be to set OTEL_DOTNET_AUTO_RESOURCE_DETECTOR_ENABLED=false to rely on [Collector's Resource Detection Processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/resourcedetectionprocessor)

Reply:
It would probably be confusing if setting the environment variable to disable resource detectors disables something that is really "required", and it would also be confusing if it did not disable all resource detectors. I wonder if we need to call out that these settings really only apply to "optional" resource detectors?

Reply:
I complete agree with the inputs shared by @pellared. `TelemetrySdk` is marked as stable in specification and this should be enabled by default. We could provide an option to turn off.

Reply:
@rajkumar-rangaraj that's what done currently. I think the question here is to enable default vs subset enabled constantly (so you cannot turn off the required ones).

I think required is satisfied already by enabling default + that leaves some room for extreme cases, where you can disable the required ones but you still know what you are doing.

Reply:
This will be handled by #2442